packages/babel-parser/src/options.ts

```diff
@@ -109,12 +109,14 @@ export function getOptions(opts?: Options | null): OptionsWithDefaults {
     options[key] = opts[key] ?? defaultOptions[key];
   }
 
-  if (
-    options.startIndex > 0 &&
-    options.startLine === 1 &&
-    typeof opts.startColumn !== "number"
-  ) {
-    options.startColumn = options.startIndex;
+  if (options.startIndex > 0 && typeof opts.startColumn !== "number") {
+    if (options.startLine === 1) {
+      options.startColumn = options.startIndex;
+    } else {
+      throw new Error(
+        "When `startIndex` is used with a `startLine` > 1 you must also provide a `startColumn`.",
+      );
+    }
```

Comment:
Inferring `startColumn` from `startIndex` looks good to me, too, as we can ship it in the next minor release.

Reply:
@JLHwung what are your thoughts on this?

```
  if (options.startColumn > 0 && opts.startIndex == null) {
    if (options.startLine === 1) {
      options.startIndex = options.startColumn;
    } else if (process.env.BABEL_8_BREAKING) {
      throw new Error(
        "When `startColumn` is used with a `startLine` > 1 you must also provide a `startIndex`.",
      );
    }
  } else if (options.startIndex > 0 && opts.startColumn == null) {
    if (options.startLine === 1) {
      options.startColumn = options.startIndex;
    } else {
      throw new Error(
        "When `startIndex` is used with a `startLine` > 1 you must also provide a `startColumn`.",
      );
    }
  }
```

* infer defaults from `startColumn` and `startIndex` when `startLine === 1` like we talked about.
* error when `startIndex` is used w/o a `startColumn` (and after line 1) since start index is new this is unconditional.
* error in babel 8 when `startColumn` is used w/o `startIndex`.

Reply:
(I pushed this up btw)

Reply:
I'm not quite sure how to have different snapshot tests for babel 8 vs babel 7 though so now the CI is failing. 😢

Reply:
You can refer to `packages\babel-parser\test\fixtures\typescript\arrow-function\async-await-null` and `packages\babel-parser\test\fixtures\typescript\arrow-function\async-await-null-babel-7`.
And update Babel 8 snapshot with `cross-env OVERWRITE=true BABEL_8_BREAKING=true yarn jest -u parser`.

Reply:
Awesome, thank you @liuxingbaoyu!

Updated the tests now 😄